.github/workflows/test_with_clone.yaml

```diff
@@ -0,0 +1,291 @@
+          SNOWFLAKE_USER: ${{ secrets.SNOWSQL_USER }}
+          SNOWFLAKE_PASSWORD: ${{ secrets.SNOWSQL_PWD }}
+
+          # Establish the account roles to be used
+          SNOWFLAKE_ADMIN_ROLE: "${{ vars.SNOWFLAKE_SYNAPSE_DATA_WAREHOUSE_DATABASE }}_ADMIN"
+
+          # Establish the domains
+          SNOWFLAKE_DOMAIN_CLONE: "${{ vars.SNOWFLAKE_SYNAPSE_DATA_WAREHOUSE_DATABASE }}_${{ github.head_ref }}"
+
+          # Establish the warehouse
+          SNOWFLAKE_WAREHOUSE: ${{ secrets.SNOWSQL_WAREHOUSE }}
+
+      steps:
+
+          - uses: actions/checkout@v4
+          - uses: actions/setup-python@v4
+            with:
+              python-version: '3.10'
+
+          - name: Configure Snowflake connections
+            run: |
+              # Make a temporary config file for the GH runner
+              config_file=$(mktemp)
+
+              # Config file for *ADMIN
+              echo 'default_connection_name = "admin"' >> $config_file
+              echo '[connections.admin]' >> $config_file
+              echo "account = \"${SNOWFLAKE_ACCOUNT}\"" >> $config_file
+              echo "user = \"${SNOWFLAKE_USER}\"" >> $config_file
+              echo "role = \"${SNOWFLAKE_ADMIN_ROLE}\"" >> $config_file
+              echo "password = \"${SNOWFLAKE_PASSWORD}\"" >> $config_file
+              echo "warehouse = \"${SNOWFLAKE_WAREHOUSE}\"" >> $config_file
+              echo 'authenticator = "SNOWFLAKE"' >> $config_file
+
+              # Write config paths to environment
+              echo "SNOWFLAKE_CONFIG_PATH=$config_file" >> $GITHUB_ENV
+
+          - name: Install Snowflake CLI
+            uses: Snowflake-Labs/snowflake-cli-action@v1.5
+            with:
+              default-config-file-path: ${{ env.SNOWFLAKE_CONFIG_PATH }}
+
+          - name: Verify Snowflake CLI installation and connections
+            run: |
+              snow --version
+              snow connection test -c admin
+
+          - name: Sanitize Clone Name
+            run: |
+              SNOWFLAKE_DOMAIN_CLONE_SANITIZED="${SNOWFLAKE_DOMAIN_CLONE//[^a-zA-Z0-9_]/_}"
+              echo "Clone name has been updated! The clone name will be: ${SNOWFLAKE_DOMAIN_CLONE_SANITIZED}"
+              echo "SNOWFLAKE_DOMAIN_CLONE_SANITIZED=${SNOWFLAKE_DOMAIN_CLONE_SANITIZED}" >> $GITHUB_ENV
+
+          - name: Drop the clone (if it exists)
+            shell: bash
+            run: |
+              snow sql -q "DROP DATABASE IF EXISTS ${SNOWFLAKE_DOMAIN_CLONE_SANITIZED};"
+
+          - name: Establish full account role name for PROXY_ADMIN
+            shell: bash
+            run: |
+              # The <CLONE>_PROXY_ADMIN account role will be used to manage interschema objects (e.g. tasks, dynamic tables)
+              # within the cloned database. First let's create the <CLONE>_PROXY_ADMIN account role name and export for immediate use:
+              echo "SNOWFLAKE_CLONE_PROXY_ADMIN_ROLE=${SNOWFLAKE_DOMAIN_CLONE_SANITIZED}_PROXY_ADMIN" >> $GITHUB_ENV
```

Comment:
Can we merge this step with the next? Just add the `SNOWFLAKE_CLONE_PROXY_ADMIN_ROLE` as a step-level environment variable.

Reply:
I prefer to keep the steps very small in scope to help with debugging, so I'd like to keep this step separate.